### PR TITLE
ref-number step

### PR DESCRIPTION
**Pull request by bendiggle**
This is for the reference number step in the form.
- Added ref-number to the step config with fields, locals and forks
- Added step fields to the fields config for input text and checkbox. The text field has a dependent on the checkbox so that it is only validated if the checkbox hasn't been checked.
- Added translations for the step label and hint
- Added the acceptance tests for the step, testing the field, label, validation and forking.
- Added the content for the step in the content file for the acceptance tests

### CODE REVIEW

File: apps/track-application/index.js [hunk 1]
```diff
@@ -39,11 +39,15 @@ module.exports = {
       }
     },
     '/ref-number': {
+      fields: ['ref-number-text', 'ref-number-checkbox'],
```
Comment (joefitter) [inline apps/track-application/index.js]: sorry to be a BALL BREAKER but can we have an item per line PLZ?

File: apps/track-application/fields/index.js [hunk 2]
```diff
@@ -28,5 +28,16 @@ module.exports = {
   'applicants-full-name': {
     mixin: 'input-text',
     validate: ['required']
+  },
+  'ref-number-checkbox': {
+  mixin: 'checkbox'
```
Comment (joefitter) [inline apps/track-application/fields/index.js]: whats up with the indentation here? mixed tabs and spaces?

Comment (bendiggle) [inline apps/track-application/fields/index.js]: good spot. I'm surprised lint didn't pick it up

**Comment by joefitter** [comment]
LGTM

**Pull request merged**